urld0f681c558_farom57_AstroidDriver_tgz-pJ8-laazotea_indi_server_examples_MinimumINDIServerJ8/src/laazotea/indi/server/INDINetworkDevice.java

```diff
@@ -0,0 +1,192 @@
+/*
+ *  This file is part of INDI for Java Server.
+ * 
+ *  INDI for Java Server is free software: you can redistribute it
+ *  and/or modify it under the terms of the GNU General Public License 
+ *  as published by the Free Software Foundation, either version 3 of 
+ *  the License, or (at your option) any later version.
+ * 
+ *  INDI for Java Server is distributed in the hope that it will be
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+ *  of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ * 
+ *  You should have received a copy of the GNU General Public License
+ *  along with INDI for Java Server.  If not, see 
+ *  <http://www.gnu.org/licenses/>.
+ */
+package laazotea.indi.server;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.ArrayList;
+import laazotea.indi.INDIException;
+
+/**
+ * A class that represent a Network Device (another INDI server).
+ *
+ * @author S. Alonso (Zerjillo) [zerjioi at ugr.es]
+ * @version 1.32, January 19, 2013
+ */
+public class INDINetworkDevice extends INDIDevice {
+
+  /**
+   * The socket to connect for the INDI Server
+   */
+  private Socket socket;
+  /**
+   * The host to connect for the INDI Server
+   */
+  private String host;
+  /**
+   * The port to connect for the INDI Server
+   */
+  private int port;
+  /**
+   * A list of names of the Device (it may be more than one)
+   */
+  private ArrayList<String> names;
+
+  /**
+   * Constructs a new Network Device and connects to it.
+   * @param server The server which listens to this Device.
+   * @param host The host to connect for the Device.
+   * @param port The port to connect for the Device.
+   * @throws INDIException if there is any problem with the connection.
+   */
+  protected INDINetworkDevice(AbstractINDIServer server, String host, int port) throws INDIException {
+    super(server);
+
+    names = new ArrayList<String>();
+
+    this.host = host;
+    this.port = port;
+
+    try {
+      socket = new Socket();
+
+      socket.connect(new InetSocketAddress(host, port), 20000);
+    } catch (IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)